packages/components/src/unit-control/utils.ts

```diff
@@ -394,7 +394,7 @@ export function filterUnitsWithSettings(
 	// Although the `isArray` check shouldn't be necessary (given the signature of
 	// this typed function), it's better to stay on the side of caution, since
 	// this function may be called from un-typed environments.
-	return Array.isArray( availableUnits )
+	return Array.isArray( availableUnits ) && Array.isArray( allowedUnitValues )
```

Comment:
I'm not too confident about this change. Because it silently squashes errors and doesn't match the expected type (`string[]`).

That said, it does fix the error when trying to change the `units` prop in Storybook. This error occurs because when you press the "Set object" button, the value is initialized as an empty object instead of an empty array.

![storybook](https://github.com/user-attachments/assets/4a69d5fd-5d32-49a8-bcb0-5ca95335c85a)

cc @WordPress/gutenberg-components

Reply:
I haven't tried, but would it make sense to change the Storybook [control type](https://storybook.js.org/docs/essentials/controls#annotation) to something like `inline-check` or `multi-select`?

Reply:
Thanks for the feedback! I've tried both controls and it feels like the `inline-check` control is the better fit.

`inline-check`;

![image](https://github.com/user-attachments/assets/280eb761-e5b7-4aec-9cfd-49b2ad9e9f3f)

`multi-select`:

![image](https://github.com/user-attachments/assets/69fd0459-2352-4995-bc06-877866f907e1)